data_tools/buildhub/utils.py

```diff
@@ -0,0 +1,62 @@
+import re
+FILE_EXTENSIONS = "zip|tar.gz|tar.bz2|dmg|apk"
+
+
+def build_record_id(record):
+    version = record["target"]["version"]
+    if record["target"]["channel"] == 'nightly':
+        url_parts = record["download"]["url"].split('/')
+        date_parts = url_parts[8].split('-')
+        date = '-'.join(date_parts[:6])
+        version = '{}_{}'.format(date, version)
+    return '{source[product]}_{version}_{target[platform]}_{target[locale]}'.format(
+        **record, version=version).replace('.', '-').lower()
+
+
+def is_release_filename(product, filename):
+    """
+    Examples of release filenames:
+
+    - firefox-53.0.tar.bz2
+
+    And things we'll want to ignore:
+
+    - firefox-1.5.0.5.tar.gz.asc
+    - firefox-52.0.win32.sdk.zip
+    """
+    re_filename = re.compile("{}-(.+)({})$".format(product, FILE_EXTENSIONS))
+    return re_filename.match(filename) and 'sdk' not in filename
+
+
+def is_release_metadata(product, version, filename):
+    """
+    Examples of release metadata filenames:
+
+    - firefox-52.0b7.json
+    - fennec-51.0b2.en-US.android-i386.json
+    """
+    re_metadata = re.compile("{}-{}(.*).json".format(product, version))
+    return re_metadata.match(filename)
+
+
+def parse_nightly_filename(filename):
+    """
+    Examples of nightly filenames:
+
+    - firefox-55.0a1.ach.win64.zip
+    - firefox-55.0a1.bn-IN.mac.dmg
+
+    And things we'll want to ignore:
+
+    - firefox-55.0a1.en-US.linux-i686.talos.tests.zip
+    - firefox-55.0a1.en-US.mac.crashreporter-symbols.zip
+    """
+    re_nightly = re.compile(r"\w+-(\d+.+)\.([a-z]+(\-[A-Z]+)?)\.(.+)\.({})$".format(
+        FILE_EXTENSIONS))
```

Comment:
nit: this line breaking is not elegant. How about splitting the regex in two parts?